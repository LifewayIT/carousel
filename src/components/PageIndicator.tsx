import React, { HTMLAttributes, ReactElement } from 'react';
import styled from 'styled-components';
import { Pages, usePageIndicator } from '../hooks/usePages';
import { cn } from '../utils/classnames';
import { device, space, color } from '../utils/styleguide';

type Props = Pages & HTMLAttributes<HTMLDivElement>;

export const PageIndicator = ({ current, total, className, ...props }: Props): ReactElement => {
  const indicators = usePageIndicator({ current, total });

  return (
    <DotContainer {...props} className={cn('lwc-page-indicator', className)}>
      {indicators.map((indicator) => (
        <Dot {...indicator} key={indicator.key} className={cn('lwc-page-dot', indicator.className)} />
      ))}
    </DotContainer>
  );
};


const DotContainer = styled.div`
  position: absolute;
  bottom: ${space._16};
  left: 0;
  right: 0;

  display: flex;
  flex-direction: row;
  align-items: center;
  justify-content: center;

  & > * {
    margin: 0 ${space._4};
  }

  @media ${device._768} {
    bottom: ${space._48};
  }
`;

const Dot = styled.div`
  width: ${space._8};
  height: ${space._8};

  border-radius: 50%;
  background-color: ${color.gray600};

  opacity: .325;

  &.lwc-current {
    opacity: 1;
  }
`;
