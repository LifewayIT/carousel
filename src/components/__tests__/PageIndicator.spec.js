import React from 'react';
import { render } from '@testing-library/react';
import { PageIndicator } from '../PageIndicator';

test('matches snapshot', () => {
  const { container } = render(<PageIndicator current={2} total={4} className="test-class" />);

  expect(container.firstChild).toMatchSnapshot();
});
