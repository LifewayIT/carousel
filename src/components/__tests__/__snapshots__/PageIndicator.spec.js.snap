// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  class="lwc-page-indicator test-class"
>
  <div
    class="lwc-page-dot"
  />
  <div
    class="lwc-page-dot lwc-current"
  />
  <div
    class="lwc-page-dot"
  />
  <div
    class="lwc-page-dot"
  />
</div>
`;
